src/abstract_charm.py

```diff
@@ -205,9 +208,13 @@ def wait_until_mysql_router_ready(self) -> None:
                 wait=tenacity.wait_fixed(5),
             ):
                 with attempt:
-                    for port in (6446, 6447):
-                        with socket.socket() as s:
-                            assert s.connect_ex(("localhost", port)) == 0
+                    if self.is_exposed():
+                        for port in (6446, 6447):
+                            with socket.socket() as s:
+                                assert s.connect_ex(("localhost", port)) == 0
+                    else:
+                        assert self._container.path("/run/mysqlrouter/mysql.sock").exists()
+                        assert self._container.path("/run/mysqlrouter/mysqlro.sock").exists()
```

Comment:
Just to be in sync. When m-r related to data-integrator, users should expose data-integrator and not m-r.

Reply:
I agree with @carlcsaposs-canonical. Isn't it ideal that mysqlrouter exposes its own ports and then unexpose when the relation with data integrator is removed? Since they are on the same machine, it is easiest if router exposes it own ports